File: Log.py
class Log:
    def authentication(self, ip, user, auth, client):
        filename = "XXX.log"
        date_format = "%Y-%m-%d %H:%M:%S"
        import datetime

        current_date = datetime.datetime.now().strftime(date_format)

        log_entry = f"\nAuthentication[{ip}][{current_date}]['{user}'][{auth}][{client}]"

        try:
            with open(filename, "a") as file:
                file.write(log_entry)
        except Exception as e:
            print("Error:", e)

File: Log.java
import java.io.FileWriter;
import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Log {
    public void authentication(String ip, String user, String auth, String client) {
        String filename = "XXX.log";
        String dateFormat = "yyyy-MM-dd HH:mm:ss";
        SimpleDateFormat sdf = new SimpleDateFormat(dateFormat);
        String date = sdf.format(new Date());

        String logEntry = String.format("\nAuthentication[%s][%s]['%s'][%s][%s]", ip, date, user, auth, client);

        try {
            FileWriter fileWriter = new FileWriter(filename, true); // true for append mode
            fileWriter.write(logEntry);
            fileWriter.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
